lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -541,6 +541,85 @@ static void getCellAsmResultNames(OpAsmSetValueNameFn setNameFn, Operation *op,
     setNameFn(op->getResult(i), prefix + portNames[i].str());
 }
 
+//===----------------------------------------------------------------------===//
+// AssignOp
+//===----------------------------------------------------------------------===//
+
+/// Returns whether the given value is a port of a component, which are
+/// defined as Block Arguments.
+static bool isComponentPort(Value value) {
+  return value.getImpl()->getKind() == detail::ValueImpl::Kind::BlockArgument;
```

Comment:
I think you can use `value.isa<BlockArgument>()`, which (through templates) boils down to the same logic.